crates/ra_syntax/src/lexer.rs

```diff
@@ -190,19 +190,24 @@ fn next_token_inner(c: char, ptr: &mut Ptr) -> SyntaxKind {
 }
 
 fn scan_ident(c: char, ptr: &mut Ptr) -> SyntaxKind {
-    let is_single_letter = match ptr.current() {
-        None => true,
-        Some(c) if !is_ident_continue(c) => true,
+    let is_raw = match (c, ptr.current()) {
+        ('r', Some('#')) => {
+            ptr.bump();
+            true
+        },
+        ('_', Some(c)) if !is_ident_continue(c) => return UNDERSCORE,
         _ => false,
     };
-    if is_single_letter {
-        return if c == '_' { UNDERSCORE } else { IDENT };
-    }
+
     ptr.bump_while(is_ident_continue);
-    if let Some(kind) = SyntaxKind::from_keyword(ptr.current_token_text()) {
+
+    if is_raw {
+        RAW_IDENT
```

Comment:
I feel slightly uneasy that adding a `RAW_IDENT` is such an invasive change. It seems like it's better to use the same `IDENT` kind for both `RAW_IDENT` and `IDENT` nodes, and differentiate raws by "does the text start with `r#`" ?

Reply:
I considered that, but decided to instead follow the precedent set by `STRING` and `RAW_STRING`